docs/InstallGuide.md

```diff
@@ -1,424 +1,358 @@
+from one guest to another.
 
-### Start VM(s)
+As a representation of the guest here we use [LXC](https://linuxcontainers.org), which helps users run multiple isolated Linux system containers. Any other virtualization technologies might be used, however we take LXC since it is easy to create/destroy/configure and simple enough to do this verification.
 
-We use kvm to start four virtual machines. Their interfaces were registered in the database earlier. These VMs and their networks are arranged as follows.
+Install LXC
 
-+ vm1, vm3 -> vnet1
-+ vm2, vm4 -> vnet2
+```
+# yum -y install lxc lxc-templates
+```
 
-The command "run-vm-all" starts all four virtual machines at once.
+Create and mount cgroup
 
-    # cd /root/vnet-test-kvm/vm
-    # ./run-vm-all
-    *** vm1 ***
-    node-id: 1
-    interface name: if-x5vjmksl
-    mac address: 52:54:00:0d:84:8c
-    *** vm2 ***
-    node-id: 2
-    interface name: if-sve1s6xp
-    mac address: 52:54:00:9a:cc:00
-    *** vm3 ***
-    node-id: 3
-    interface name: if-bvixz40n
-    mac address: 52:54:00:0a:fb:78
-    *** vm4 ***
-    node-id: 4
-    interface name: if-7st2alal
-    mac address: 52:54:00:e5:f9:fb
+```
+# mkdir /cgroup
+# echo "cgroup /cgroup cgroup defaults 0 0" >> /etc/fstab
+# mount /cgroup
+```
 
-### Check the IP address of virtual machines
+Create 2 LXC guests
+(it may require rsync to be installed)
 
-You can log into the virtual machines using the telnet command.
+```
+# lxc-create -t centos -n inst1
+# lxc-create -t centos -n inst2
+```
 
-    # telnet localhost PORT-NO
+Configure interfaces of each guest
 
-The port numbers are:
+```
+# vi /var/lib/lxc/inst1/config
+lxc.network.type = veth
+lxc.network.flags = up
+lxc.network.veth.pair = inst1
+lxc.network.ipv4 = 10.100.0.10
+lxc.network.hwaddr = 10:54:FF:00:00:01
+lxc.rootfs = /var/lib/lxc/inst1/rootfs
+lxc.include = /usr/share/lxc/config/centos.common.conf
+lxc.arch = x86_64
+lxc.utsname = inst1
+lxc.autodev = 0
+```
 
-+ vm1 : 5001
-+ vm2 : 5002
-+ vm3 : 5003
-+ vm4 : 5004
+```
+# vi /var/lib/lxc/inst2/config
+lxc.network.type = veth
+lxc.network.flags = up
+lxc.network.veth.pair = inst2
+lxc.network.ipv4 = 10.100.0.11
+lxc.network.hwaddr = 10:54:FF:00:00:02
+lxc.rootfs = /var/lib/lxc/inst2/rootfs
+lxc.include = /usr/share/lxc/config/centos.common.conf
+lxc.arch = x86_64
+lxc.utsname = inst2
+lxc.autodev = 0
+```
 
-The ip addresses are assigned as follows. As you can see, vm1 and vm3, share the same ip address, as do vm2 and vm4.
+Make sure that the IPv4 address and MAC address are the same as what you specify when you create the interface database records.
 
-+ vm1 : 10.102.0.10
-+ vm2 : 10.102.0.10
-+ vm3 : 10.102.0.11
-+ vm4 : 10.102.0.11
+Launch the LXC guests then enslave the LXC's tap interfaces to the datapath.
 
-Log into the virtual machines and make sure their ip addresses are correct.
+```
+# lxc-start -d -n inst1
```

Comment:
I get the following result after this command.

```
[15:09:49] centos@joske  (~) > sudo lxc-start -d -n inst1
lxc-start: lxc_start.c: main: 342 The container failed to start.
lxc-start: lxc_start.c: main: 344 To get more details, run the container in foreground mode.
lxc-start: lxc_start.c: main: 346 Additional information can be obtained by setting the --logfile and --logpriority options.



[15:11:42] centos@joske  (~) > sudo lxc-start -n inst1
lxc-start: conf.c: lxc_mount_auto_mounts: 801 No such file or directory - error mounting proc on /usr/lib64/lxc/rootfs/proc flags 14
lxc-start: conf.c: lxc_setup: 4153 failed to setup the automatic mounts for 'inst1'
lxc-start: start.c: do_start: 688 failed to setup the container
lxc-start: sync.c: __sync_wait: 51 invalid sequence number 1. expected 2
lxc-start: start.c: __lxc_start: 1080 failed to spawn 'inst1'
lxc-start: lxc_start.c: main: 342 The container failed to start.
lxc-start: lxc_start.c: main: 346 Additional information can be obtained by setting the --logfile and --logpriority options.
```

Do these commands work on a minimal centos installation? I think there might be a problem with LXC's Centos template. That would make these commands work on machines where an old working template is installed but not on newly installed machines that will download a fresh template.